### DISCUSSION TITLE
How do I create an app with my own TensorFlow model

### DISCUSSION THREAD

**Discussion by MMelQin** [Project-MONAI/monai-deploy-app-sdk#339] [Show and tell]
For now, wrap the inference in an operator implementing the base Operator class or even the InferenceOperator class, with the model embedded at static content in the App itself.

The model file itself needs to be within the App folder itself, so that it will be picked up by the moai-deploy Packager on packaging the whole app folder; this also meanings that the model path need not be provided on the Packager command line as for the `-m` option.
At runtime, the relative file path, or path from the App root, needs to be accessed by the inference operator to load the network. Avoid using the model factory, as the factory would have no models to load.

There are ways to avoid having the model in the path of the App folder, and then provided on running Packager to inject into the MONAI App Package, but this will require setting the environment variables on running the MAP Docker. Will go in to this a little more later.

**Comment by slbryson** [comment]
This is a great way to test wrapping a TensorFlow based model in a custom operator to reuse a proven or approved model inference in the MONAI Framework. In the documentation [Create Operator Classes](https://docs.monai.io/projects/monai-deploy-app-sdk/en/latest/developing_with_sdk/creating_operator_classes.html), a developer can follow the examples in that section including the example [MONAISegInferenceOperator](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.monai.io%2Fprojects%2Fmonai-deploy-app-sdk%2Fen%2Flatest%2F_modules%2Fmonai%2Fdeploy%2Foperators%2Fmonai_seg_inference_operator.html%23MonaiSegInferenceOperator&data=05%7C01%7Csbryson%40nvidia.com%7Cb46647632e014cccf92508da92ad266b%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C637983569613781634%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=jDNoBLUEOnpJgERj7yRCjFB0CcbeZdolIg2y%2F5MCUzk%3D&reserved=0) that inherits [InferenceOperator](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.monai.io%2Fprojects%2Fmonai-deploy-app-sdk%2Fen%2Flatest%2F_modules%2Fmonai%2Fdeploy%2Foperators%2Finference_operator.html%23InferenceOperator&data=05%7C01%7Csbryson%40nvidia.com%7Cb46647632e014cccf92508da92ad266b%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C637983569613781634%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=OBzH%2B9LL0RDmduY1O1b%2FICzF%2FMMBQjMPE4HTPL8k8NA%3D&reserved=0).
